modules/peers.js

```diff
@@ -62,232 +56,311 @@ __private.attachApi = function () {
-	var where = [];
-	var params = {};
-
-	if (filter.port) {
-		where.push('"port" = ${port}');
-		params.port = filter.port;
-	}
-
-	if (filter.state >= 0) {
-		where.push('"state" = ${state}');
-		params.state = filter.state;
-	}
-
-	if (filter.os) {
-		where.push('"os" = ${os}');
-		params.os = filter.os;
-	}
-
-	if (filter.version) {
-		where.push('"version" = ${version}');
-		params.version = filter.version;
-	}
-
-	if (filter.broadhash) {
-		where.push('"broadhash" = ${broadhash}');
-		params.broadhash = filter.broadhash;
-	}
-
-	if (filter.height) {
-		where.push('"height" = ${height}');
-		params.height = filter.height;
-	}
-
-	library.db.query(sql.countByFilter({
-		where: where
-	}), params).then(function (rows) {
-		var res = rows.length && rows[0].count;
-		return setImmediate(cb, null, res);
-	}).catch(function (err) {
-		library.logger.error(err.stack);
-		return setImmediate(cb, 'Peers#count error');
+__private.insertSeeds = function (cb) {
+	var updated = 0;
+	library.logger.trace('Peers->insertSeeds');
+	async.each(library.config.peers.list, function (peer, eachCb) {
+		peer = library.logic.peers.create(peer);
+		library.logger.trace('Processing seed peer: ' + peer.string);
+		self.ping (peer, function (err) {
+			++updated;
+			return setImmediate(eachCb);
+		});
+	}, function (err) {
+		library.logger.trace('Peers->insertSeeds - Peers discovered', {updated: updated, total: library.config.peers.list.length});
+		setImmediate(cb);
 	});
 };
 
-__private.banManager = function (cb) {
-	library.db.query(sql.banManager, { now: Date.now() }).then(function (res) {
-		return setImmediate(cb, null, res);
+__private.dbLoad = function (cb) {
+	var updated = 0;
+	library.logger.trace('Importing peers from database');
+	library.db.any(sql.getAll).then(function (rows) {
+		library.logger.trace('Imported peers from database', {count: rows.length});
+		async.each (rows, function (peer, eachCb) {
+			peer = library.logic.peers.create(peer);
+
+			if (library.logic.peers.exists(peer)) {
+				peer = library.logic.peers.get(peer);
+				if (peer && peer.state > 0 && Date.now() - peer.updated > 3000) {
+					self.ping (peer, function (err) {
+						++updated;
+						return setImmediate(eachCb);
+					});
+				} else {
+					return setImmediate(eachCb);
+				}
+			} else {
+				self.ping (peer, function (err) {
+					++updated;
+					return setImmediate(eachCb);
+				});
+			}
+		}, function (err) {
+			library.logger.trace('Peers->dbLoad Peers discovered', {updated: updated, total: rows.length});
+			return setImmediate(cb);
+		});
 	}).catch(function (err) {
-		library.logger.error(err.stack);
-		return setImmediate(cb, 'Peers#banManager error');
+		library.logger.error('Import peers from database failed', {error: err.message || err});
+		return setImmediate(cb);
 	});
 };
 
-__private.getByFilter = function (filter, cb) {
-	var where = [];
-	var params = {};
+__private.dbSave = function (cb) {
+	var peers = library.logic.peers.list();
```

Comment:
We should call normalized version `library.logic.peers.list(true);`, so later not need to set default values for SQL query - will be more readable.